test/views/returns/credits/ConvertedCreditsWeightViewSpec.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.returns.credits
+
+import base.ViewSpecBase
+import forms.returns.credits.ConvertedCreditsWeightFormProvider
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.mockito.MockitoSugar
+import org.mockito.integrations.scalatest.ResetMocksAfterEachTest
+import play.api.i18n.Messages
+import play.api.mvc.Call
+import play.twirl.api.Html
+import support.{ViewAssertions, ViewMatchers}
+import uk.gov.hmrc.scalatestaccessibilitylinter.AccessibilityMatchers
+import views.html.returns.credits.ConvertedCreditsWeightView
+
+class ConvertedCreditsWeightViewSpec extends ViewSpecBase
+  with ViewAssertions
+  with ViewMatchers
+  with AccessibilityMatchers
+  with MockitoSugar {
+
+  override val messages = spy(super.messages)
+  private val form = new ConvertedCreditsWeightFormProvider()()
+  private val page = inject[ConvertedCreditsWeightView]
+  private def createView: Html = page(form, Call("method", "/submit-url"))(request, messages)
+
+  "AnotherBusinessExportWeightView" should {
+    
+    // Note as this only runs once, no mocks are reset
+    val view = createView
+
+    "have a title and heading" in {
+      verify(messages, times(2)).apply("converted-credits-weight.heading-title")
+      view.select("title").text must include ("Submit return - Plastic Packaging Tax - GOV.UK")
```

Comment:
should we test here if we get the right full text? I take these test as integration test, that load the app and test how components interact with the view